### DISCUSSION TITLE
Can we package the mtex into an application, which can work in a PC without matlab?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#2195] [Ask Anything] [answered]
Hi, everyone,
     I am trying to make some matlab applications with the GUI interface, based on the mtex. It works well.
     Now I want to make those applications work in PC without matlab. I have tried the "Application compiler" function to package the application into a "exe" file. However, many errors occur when I run the app. It seems like that the mtex codes and functions can not be correctly packaged.
     Does anyone know the proper way to do this job?
     Thanks!
     Brian
![image](https://github.com/user-attachments/assets/106eefb2-1b8c-4a51-8ccd-81b243a5d6c5)

**Comment by Brian-Liu-rx** [comment]
Here is one error. 
![image](https://github.com/user-attachments/assets/7c127b0c-82b2-4d8a-b584-f814fee1dcd4)

**Comment by nyyssont** [comment]
I think this has been tried every once in a while, so far without success or understanding of the underlying issues. I suspect heavy modifications to MTEX would be required, anyway.

**Comment by Brian-Liu-rx** [comment]
Still looking for help！

**Comment by ralfHielscher** [comment, marked as answer]
I think it should be manageable to get MTEX running as an independent app. For the specific error you are experiencing it would be helpful to remove the `try / catch` statement in `loadEBSD_generic` to see the actual error message.

**Reply by Brian-Liu-rx** [reply]
Hi, Ralf,
    Thanks for your reply!
    The error mentioned above is just taken as an example. Even though it was solved, another error may then occur.
    I notice that in Mtex, the subroutines or funtions (.m files) are categorized and placed in different folders. For example, when we run the code "orientation.byEuler()", the system will find the folder named "@orientation" first and then run the "byEuler.m" file in that folder.  
    However, when we do the  "Application compiler" , all the subroutines (.m files) will be found and packaged together. The code "orientation.byEuler()" can not be run correctly in the independent app.
    Ralf, have you tried to make some apps based on mtex using "Application compiler" before?

    Thanks!
     Brian

**Reply by Brian-Liu-rx** [reply]
still looking for help

**Comment by Brian-Liu-rx** [comment]
still looking for help

**Reply by xiu-shen** [reply]
Have your problems been solved? Could you please offer some suggestions?
